src/analysis/convex/cone/proper.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2022 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import analysis.inner_product_space.adjoint
+
+/-!
+
+We define a proper cone as a nonempty, closed, convex cone. Proper cones are used in defining conic
+programs which generalize linear programs. A linear program is a conic program for the positive
+cone. We then prove Farkas' lemma for conic programs following the proof in the reference below.
+Farkas' lemma is equivalent to strong duality. So, once have the definitions of conic programs and
+linear programs, the results from this file can be used to prove duality theorems.
+
+## References
+- [B. Gartner and J. Matousek, Cone Programming ][gartnerMatousek]
+
+-/
+
+open continuous_linear_map filter
+
+namespace convex_cone
+
+variables {E : Type*} [normed_add_comm_group E] [inner_product_space ℝ E]
```

Comment:
Those assumptions look too strong.

Reply:
Thanks! I'll make these weaker. But I might need to add these two back somewhere later when I add in other lemmas.

Reply:
Then I guess you'll need (at least) two sections, which is okay.